gno.land/pkg/sdk/vm/keeper.go

```diff
@@ -1246,14 +1243,14 @@ func logTelemetry(
 	// Record the CPU cycles
 	metrics.VMCPUCycles.Record(
 		context.Background(),
-		cpuCycles,
+		gasUsed.CategoryDetails()["CPU"].Total.GasConsumed,
```

Comment:
`CategoryDetails` is used to read the CPU total, but this method rebuild the entire map. So it seems like the map construction costs and memory allocation would occur every time the `logTelemetry` function is called.

Reply:
Yes, indeed, but since this function is called only once per maketx addPkg / call / run, it seems okay to me to build the map at that moment so we keep the metering as fast as possible.